docs/src/main/sphinx/connector/filesystem-cache.md

```diff
@@ -0,0 +1,147 @@
+# Filesystem cache
+
+Trino accesses files directly on object storage and remote filesystem storage.
+This often involves the transfer of large amounts of data. The files are
+retrieved from HDFS, or any other supported object storage, by multiple workers
+and processed on these workers. Repeated queries with different parameters, or
+even different queries from different users, often access, and therefore
+transfer, the same objects.
+
+Trino includes support for caching these files with the help of the open
+source [Alluxio](https://github.com/Alluxio/alluxio) libraries with catalogs
+using the following connectors:
+
+* [](/connector/delta-lake)
+
+(fs-cache-distributed)=
+## Distributed caching
```

Comment:
I added this whole new section @raunaqmorarka @jkylling @Pluies @electrum to cover more details on what is happening.

Reply:
This looks great! Added some suggestions below.